mantidproject/mantid
# Pull nexus library into mantid

This pulls in the minimum amount of code from https://github.com/nexusformat/code/ to replace the nexus dependency in mantid. The criteria for success is that the nexus library is no longer a dependency and the source is changed enough to compile without warnings, pass cppcheck, and all tests pass. There is a lot of work left to do to make the code performant and bug free.

Refs #38332 and [EWM8446](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=8446)



#### Further detail of work

Random comment: Changing `.c` to `.cpp` put cmake into a mode where it now acknowledges that these are c++ files and more changes had to be made.

Much of the information on how to build nexus was taken from the [nexus conda package](https://github.com/conda-forge/nexus-feedstock/blob/main/recipe/meta.yaml) which is [v4.4.3 of nexus](https://github.com/nexusformat/code/releases/tag/v4.4.3) (Aug 15, 2016) despite there being changes as recent as Jan 26, 2020. The code from the most recent changes, rather than the tag, was the starting point.



### To test:

This build will check that system tests and packaging work on all OS:
https://builds.mantidproject.org/job/build_packages_from_branch/961/








---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
